### ISSUE TITLE
[Optimization] I tried to find the optimization using caputol1method

### ISSUE THREAD

**Issue by Velmurugan1008** [alexfikl/pycaputo#33]
import numpy as np
import matplotlib.pyplot as plt
from pycaputo.utils import Array
from pycaputo import CaputoDerivative, CaputoL1Method, Side, diff
from pycaputo.grid import make_uniform_points
from pycaputo.utils import figure, set_recommended_matplotlib

# Define the function
def f(x: Array) -> Array:
    return (x - 5) ** 2

# Set up Caputo derivative and method
d = CaputoDerivative(order=0.9, side=Side.Left)
method = CaputoL1Method(d)

# Generate points for plotting and calculating numerical Caputo derivative
p = make_uniform_points(256, a=0.0, b=2.0)
df_num = np.array([diff(method, f, np.array([x]))[0] for x in p.x])

# Plot the Caputo derivative using the L1 method
set_recommended_matplotlib()
with figure("caputo-derivative-l1.svg") as fig:
    ax = fig.gca()
    ax.plot(p.x, df_num, lw=5, label="$L1~ Method$")
    ax.set_xlabel("$x$")
    ax.set_ylabel(r"$D^\alpha_C[f](x)$")
    ax.legend()
    plt.show()

# Initial setup for optimization using CaputoL1Method
x = 0
learning_rate = 0.1
error = []

plt.plot(p.x, df_num, lw=5, label="$L1~ Method$")
plt.xlabel('X')
plt.ylabel(r"$D^\alpha_C[f](x)$")

for i in range(25):
    # Use CaputoL1Method for optimization
    grad = diff(method, f, np.array([x]))[0]
    x = x - learning_rate * grad
    y = (x - 5) ** 2
    error.append(y)
    plt.scatter(x, y)

    # Print the values corresponding to each iteration
    print(x, grad)
    
plt.show()
I have some errors while finding the fractional gradient descent. Please help me to solve this issue.
Thank you.

**Comment by alexfikl** [comment]
Is this based one some generalized Taylor formula like https://doi.org/10.1016/j.amc.2006.07.102?

What sort of errors are you getting? Does just running the example in `examples/caputo-derivative-l1.py` work? If not, you may be missing some dependencies or the package isn't installed correctly. I haven't gotten to publishing this on PyPI, so it might be a bit annoying to install..

**Comment by alexfikl** [comment]
Ah, I see what you want to do! That requires some work to modify the current interface, but I agree that it's necessary.

I'll probably get to work on it over the winter holiday.

**Comment by alexfikl** [comment]
@Velmurugan1008 I added a wrapper (and an example) to compute the gradient at a given set of points only in #35. It's probably not as fast as it could be, so it will need more work if you want to scale.

You can use it as inspiration to make your own wrapper, if nothing else. Feel free to open more issues or pull requests about this!
